### PR TITLE
s3 upload v3

### PR DESCRIPTION
**Pull request by marcellmueller**
Implements # \<issue number\>

### CODE REVIEW

File: app/backend/lib/graphql/resolveFileUpload.ts [hunk 1]
```diff
@@ -34,37 +36,47 @@ if (!isDeployedToOpenShift) {
...
 
-  return s3Upload;
+    parallelUploads3.on('httpUploadProgress', (progress) => {
+      console.log(progress);
+    });
+
+    const data = await parallelUploads3.done();
+    console.log(data);
+
+    const key = (data as CompleteMultipartUploadCommandOutput).Key;
```
Comment (wenzowski) [inline app/backend/lib/graphql/resolveFileUpload.ts:69]: what should be happening here if data isn't of the type we're casting to?

File: app/backend/lib/graphql/resolveFileUpload.ts [hunk 2]
```diff
@@ -34,37 +36,47 @@ if (!isDeployedToOpenShift) {
...
+      partSize: 1024 * 1024 * 5, // optional size of each part, in bytes, at least 5MB
+      leavePartsOnError: false, // optional manually handle dropped parts
     });
 
-  return s3Upload;
+    parallelUploads3.on('httpUploadProgress', (progress) => {
+      console.log(progress);
+    });
+
+    const data = await parallelUploads3.done();
```
Comment (wenzowski) [inline app/backend/lib/graphql/resolveFileUpload.ts:66]: we confirmed that by awaiting done all parts should be uploaded at this point

File: app/backend/lib/graphql/resolveFileUpload.ts [hunk 3]
```diff
@@ -34,37 +36,47 @@ if (!isDeployedToOpenShift) {
...
+    console.log(data);
+
+    const key = (data as CompleteMultipartUploadCommandOutput).Key;
+    if (!key) {
+      throw new Error('Data does not contain a key');
```
Comment (wenzowski) [inline app/backend/lib/graphql/resolveFileUpload.ts:71]:
```suggestion
      throw new Error('Data does not contain a key', data);
```

File: app/backend/lib/s3client.ts [hunk 4]
```diff
@@ -25,3 +31,18 @@ aws.config.update({
 const s3Client = new aws.S3();
 
 export default s3Client;
```
Comment (wenzowski) [inline app/backend/lib/s3client.ts:33]: tech debt on leaving the v2 sdk for other processes needs to be tracked

File: app/backend/lib/s3client.ts [hunk 5]
```diff
@@ -25,3 +31,18 @@ aws.config.update({
...
+
+const awsConfig = {
+  region: AWS_S3_REGION,
+  credentials: fromTemporaryCredentials({
+    masterCredentials: fromEnv(),
```
Comment (wenzowski) [inline app/backend/lib/s3client.ts:38]: can we add a comment here noting which env vars `fromEnv()` is pulling?

File: app/backend/lib/s3client.ts [hunk 6]
```diff
@@ -25,3 +31,18 @@ aws.config.update({
...
+    params: {
+      RoleArn: AWS_ROLE_ARN,
+      RoleSessionName: `s3-v3-role-session-${Date.now()}`,
+      DurationSeconds: 3600,
```
Comment (wenzowski) [inline app/backend/lib/s3client.ts:42]: we confirmed that when the temporary credentials expire this factory is documented to use the master credentials to refresh the temporary

File: app/tests/backend/lib/graphql/resolveFileUpload.test.ts [hunk 7]
```diff
@@ -12,7 +12,7 @@ describe('The saveRemoteFile function', () => {
 
   it('should throw an error with unsupported payload', async () => {
     await expect(saveRemoteFile({})).rejects.toThrow(
-      'Error, unable to upload to S3: Error: Unsupported body payload object'
+      'Unsupported body payload object'
     );
   });
```
Comment (wenzowski) [inline app/tests/backend/lib/graphql/resolveFileUpload.test.ts:17]: tech debt: missing test for broken pipe; missing test for checksummed roundtrip

File: .github/workflows/main.yaml [hunk 8]
```diff
@@ -26,7 +26,7 @@ jobs:
       RENOVATE_PRIVATE_KEY: ${{ secrets.RENOVATE_PRIVATE_KEY }}
 
   deploy:
-    if: github.event.ref == 'refs/heads/main'
+    # if: github.event.ref == 'refs/heads/main'
```
Comment (wenzowski) [inline .github/workflows/main.yaml:29]: must revert before merging

**Review by wenzowski — CHANGES_REQUESTED** [review]
so close!

File: app/backend/lib/s3client.ts (hunk 4 above)
Comment (AntBush) [reply app/backend/lib/s3client.ts:33]: https://app.zenhub.com/workspaces/ccbc---delivery-board-61c0c1204c8bcb001491c5f3/issues/bcgov/conn-ccbc-portal/981

File: app/tests/backend/lib/graphql/resolveFileUpload.test.ts [hunk 9]
```diff
@@ -12,7 +16,7 @@ describe('The saveRemoteFile function', () => {
 
   it('should throw an error with unsupported payload', async () => {
     await expect(saveRemoteFile({})).rejects.toThrow(
```
Comment (AntBush) [inline app/tests/backend/lib/graphql/resolveFileUpload.test.ts:18]: A bit late, but shouldn't this be: 
```suggestion
    await expect(saveRemoteFile(undefined)).rejects.toThrow(
```
?

**Review by wenzowski — APPROVED** [review]

**Pull request merged**